File: Medium/1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        n=len(arr)

        arr.sort()
        arr[0]=1
        res=0
        i=1
        while i<n:
            if arr[i]-arr[i-1]>1:
                arr[i]=arr[i-1]+1
            i+=1

        return arr[n-1]

File: Medium/1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.java
class Solution{
    public int maximumElementAfterDecrementingAndRearranging(int[] arr){
        int n=arr.length;

        Arrays.sort(arr);
        arr[0]=1;
        int res=0;
        int i=1;
        while(i<n){ 
            if(arr[i]-arr[i-1]>1){
                arr[i]=arr[i-1]+1;
            }
            i++;    
        }

        return arr[n-1];

    }
}